src/Stratis.Bitcoin.Features.PoA/PoANetwork.cs

```diff
@@ -168,7 +161,7 @@ public PoANetwork()
             Assert(this.Consensus.HashGenesisBlock == uint256.Parse("0x0621b88fb7a99c985d695be42e606cb913259bace2babe92970547fa033e4076"));
             Assert(this.Genesis.Header.HashMerkleRoot == uint256.Parse("0x9928b372fd9e4cf62a31638607344c03c48731ba06d24576342db9c8591e1432"));
 
-            if ((this.FederationPublicKeys == null) || (this.FederationPublicKeys.Count == 0))
+            if ((this.ConsensusOptions.FederationPublicKeys == null) || (this.ConsensusOptions.FederationPublicKeys.Count == 0))
```

Comment:
You can bring this almost all the way to the top of the class. Maybe just after `var consensusOptions = new PoAConsensusOptions`

Reply:
not sure. 
network initialize has parts- first is setup and second is sanity checks. 
I think it's a good idea to avoid mixing them

Reply:
We don't really need the sanity checks, but it's a shame to go through the whole method only to fail because some things weren't set up properly.
Even the genesis checks are a bit useless in my opinion.

Reply:
ok fine